### PR TITLE
[IMP] fieldservice_activity: Improved code

### PR DESCRIPTION
**Pull request by murtuzasaleh**

### CODE REVIEW

**Comment by murtuzasaleh** [comment]
@max3903

File: fieldservice_activity/tests/test_fsm_activity.py [hunk 1]
```diff
@@ -67,3 +68,21 @@ def get_activity_vals(self, name, required, ref, order_id):
...
+                 'ref': 'Ref new',
+                 'state': 'todo',
+             })]})
+        # Create an Order
+        self.fso = self.Order.create({
+            'location_id': self.test_location.id,
+            'template_id': self.template.id,
+        })
+        # Test _onchange_template_id()
+        self.fso._onchange_template_id()
```
Comment (max3903) [inline fieldservice_activity/tests/test_fsm_activity.py:88]: @murtuzasaleh Can you check that the order has 1 activity and the activity is different from the one related to the template?

**Review by max3903 — APPROVED** [review]

**Comment by brian10048** [comment]
Is it the intended result to append new activities when changing the template?

Testing on runbot, I changed the template and received an appended list of activities for the previous template and the new one.  I expected to receive only the activities for the new template.

**Comment by max3903** [comment]
@brian10048 We can't really flush all the existing activities on the orders. You may have added some manually.

**Comment by brian10048** [comment]
@max3903 

I tested on this runbot build.  Activities not added manually, but through the onchange of template_id

Configure 2 activities (1 & 2) on Template A
Configure 2 different activities (3 & 4) on Template B

Create a new order. Set to Template A
Activities 1 & 2 populate

Change template to B
Activities 3 & 4 added with 1 & 2 remaining.


Is it an issue to first delete all activities and then repopulate the activities from the new template?

File: fieldservice_activity/models/fsm_order.py [hunk 2]
```diff
@@ -11,14 +11,22 @@ class FSMOrder(models.Model):
...
     def _onchange_template_id(self):
         res = super()._onchange_template_id()
-        if self.template_id:
-            for temp_activity_id in self.template_id.temp_activity_ids:
-                activity_id = temp_activity_id
-                activity_id.fsm_template_id = False
-                activity_id.fsm_order_id = self
+        for rec in self:
+            if rec.template_id:
+                activity_list = []
```
Comment (brian10048) [inline fieldservice_activity/models/fsm_order.py:20]:
```suggestion
                activity_list = []
                rec.order_activity_ids = (5, 0, 0)
```
@max3903 something like this?

**Comment by max3903** [comment]
@wolfhall @rto-odoo What do you guys think?

**Review by nelsonramirezs — APPROVED** [review]
LGTM

**Review by osi-scampbell — APPROVED** [review]

**Comment by brian10048** [comment]
@max3903 
> @wolfhall @rto-odoo What do you guys think?

Any further thoughts about deleting the previous activity lines before adding new ones from the new template?

**Comment by max3903** [comment]
@brian10048 I don't think we can blindly delete all the lines. It is safer to let the user decides.

**Review by marcelsavegnago — APPROVED** [review]

**Comment by max3903** [comment]
@sbidoul Can you help us here? Looks like @OCA-git-bot is stuck on something.

**Pull request merged**